pages/api/roles/index.ts

```diff
@@ -143,6 +122,14 @@ async function handlePost(req: NextApiRequest, res: NextApiResponse<ApiResponse<
   res.status(HttpStatus.CREATED).json(createJsonResponse(newRole, messages[MessageType.ROLE_CREATED_SUCCESSFULLY]));
 }
 
+function parseGetQueryParams(req: NextApiRequest): RoleQueryParams {
+  const { companyId, searchQuery } = req.query;
+  const searchWords =
+    searchQuery === undefined ? [] : Array.isArray(searchQuery) ? searchQuery : searchQuery.trim().split(/\s+/);
```

Comment:
Are we matching word by word? 

What if searchWords is an array like ["software engineer", "site re"]?

Reply:
For our use search words is supposed to be a string and each item in the array should be considered individual units to compare against if we want to support other cases in future.

I think it's not worth to split words in the array for now, although we can do it.

Reply:
Lets save time by not splitting the words in the array for now.